### ISSUE TITLE
[FEATURE REQUEST]

### ISSUE THREAD

**Issue by SpEcHiDe** [rg3/youtube-dl#6030]
please include an auto update option like Google Chrome which will notify users that an update is available instead of manually checking for an update.

This can be achieved by checking for the version information whenever youtubedl is run to download some video.

**Comment by vxbinaca** [comment]
Each platform of install has a problem with this:

Repository: The problem I see with this is you'd need to tailor a special version of youtube-dl for OS repositories that didn't do this automatically so as not to break apt-yum et al when uninstalling it or creating crazy deps like git or pip. Why not just use pip at that point?

To be fair: Steam works with an initial install thats then updated from Valve.

Pip: Same as apt/yum, but the maintainers are close to the git version, while the Ubuntu version seems to lag behind until it breaks for youtube and enough people complain and then it's fixed.

Manual: It would work, sort of. We'd have to have the script call su so it could change thats in /use/local/bin.


So why not just use the pip version, or manually install youtube-dl (after cleanly removing any other version installed by another method, I just learned this hard lesson), and then using the manually installed one and every now and again calling -U as sudo?

**Comment by vxbinaca** [comment]
This would create confusion for people who used methods other than manual installation (eg apt/pip).

**Comment by SpEcHiDe** [comment]
How about something like this ?
civ = get_current_installed_version()
cuv = get_latest_upstream_version()
if civ != cuv : 
  pynotify("update is available")

**Comment by vxbinaca** [comment]
It's best package maintainers actually make their stuff somewhat current and not let it become stale, like Ubuuntu did (it's maintainer let it go for 6 months), or have users rely on manually installed versions. I'm stricktly speaking about Linux/OSX users.

**Comment by SpEcHiDe** [comment]
Can I work on this issue? @vxbinaca

**Comment by vxbinaca** [comment]
@SpEcHiDe Yes if you'd like. Call home does something similar but what's needed is someone - not me I have my own [responsibilities] (https://github.com/bibanon/tubeup) - to keep the Ubuntu version somewhat up to date with Pip. To where the major sites, not all but the most used ones, work. The maintainer only appears to update the script when Youtube stops working and the versions drift really far after a while.

I don't worry about this as I just use Pip. My script installs from Pip and pulls youtube-dl accordingly from Pip.

I'm worried about the novices who aren't skilled in terms of easing the archival/mirroring work they do.
